### ISSUE TITLE
upgrade from pysam-0.9.1.2 to pysam-0.11.2.2 fails

### ISSUE THREAD

**Issue by splaisan** [pysam-developers/pysam#495]
Attached the full terminal log during upgrade  (triggered by installing CrossMap)

Thanks for your help, this error is now on two independent macs with different OS'es (Yosemite and ElCapitan). lzma.h seem to be missing!! but lzma is installed and the .h is present in my macport include folder

```
ll /opt/local/include/lzma.h 
-rw-r--r-- 1 root admin 9.6K Dec 31  2016 /opt/local/include/lzma.h
```

```
lzma --version
xz (XZ Utils) 5.2.3
liblzma 5.2.3
```

Best

[pysam_upgrade_failed.txt](https://github.com/pysam-developers/pysam/files/1117873/pysam_upgrade_failed.txt)

**Comment by splaisan** [comment]
I now tried to specify the extra include folder with

`sudo -H pip install --global-option=build_ext --global-option="-I/opt/local/include/" pysam -I 2>&1 | tee -a ~/Desktop/pysam-install_log.txt`

which failed again but differently

[pysam-install_log.txt](https://github.com/pysam-developers/pysam/files/1118928/pysam-install_log.txt)

Any idea?

**Comment by splaisan** [comment]
I found that I had /usr/local/lib/liblzma.5.dylib but it is not named  lzma.h and is not found.
Can I create some softlink to it to make the installer happy or is liblzma.5.dylib something different from  lzma.h ?
I would really like to fix my system for all python packages that require and do not find this header.

**Comment by AndreasHeger** [comment]
Hi, it might be that you need the development libraries (usually called -devel or similar)? If only the compiled libraries are installed you are able use tools with liblzma support, but without any of the header files required you will not be able to compile new tools.

**Comment by AndreasHeger** [comment]
on RH systems you would do: yum install lzma-devel

**Comment by splaisan** [comment]
I ended up building the full https://tukaani.org/xz/ from source which installed lzma folder and lzma.h in /usr/local/include/

My installed CrossMap.py still complains but not anymore about lzma. Note that this is not the case under virtualenv as pointed elsewhere. It seems that the standard install lacks some substitution of @rpath (if my understanding is right!)

```
CrossMap.py
Traceback (most recent call last):
  File "/usr/local/bin/CrossMap.py", line 22, in <module>
    import pysam
  File "/Library/Python/2.7/site-packages/pysam/__init__.py", line 5, in <module>
    from pysam.libchtslib import *
  File "pysam/libchtslib.pyx", line 1, in init pysam.libchtslib (pysam/libchtslib.c:12515)
ImportError: dlopen(/Library/Python/2.7/site-packages/pysam/libcutils.so, 2): Library not loaded: @rpath/pysam-0.11.2.2-py2.7-macosx-10.10-intel.egg/pysam/libcsamtools.so
  Referenced from: /Library/Python/2.7/site-packages/pysam/libcutils.so
  Reason: image not found
```

```
ll /Library/Python/2.7/site-packages/pysam/libcutils.so
-rwxr-xr-x 1 root wheel 264K Jul  3 12:43 /Library/Python/2.7/site-packages/pysam/libcutils.so
```

All Chinese to me :-)

Thanks

**Comment by AndreasHeger** [comment]
I am running into a similar issue now with Travis CI.
```
E   ImportError: dlopen(/Users/travis/miniconda3/envs/testenv/lib/python2.7/site-packages/pysam-0.12.0.1-py2.7-macosx-10.5-x86_64.egg/pysam/libchtslib.so, 2): Library not loaded: @rpath/liblzma.5.dylib

E     Referenced from: /Users/travis/miniconda3/envs/testenv/lib/python2.7/site-packages/pysam-0.12.0.1-py2.7-macosx-10.5-x86_64.egg/pysam/libchtslib.so

E     Reason: Incompatible library version: libchtslib.so requires version 8.0.0 or later, but liblzma.5.dylib provides version 6.0.0[0m
```
This used to work fine and stopped working recently after an unrelated code change.

**Comment by AndreasHeger** [comment]
Note that this is only for python2.7 on OSX, py3 on OSX works fine.

**Comment by nieder** [comment]
~~~~
Reason: Incompatible library version: libchtslib.so requires version 8.0.0 or later, but liblzma.5.dylib provides version 6.0.0
~~~~
You linked libchtslib.so against liblzma.5.dylib that was at soversion 8.0.0, but at runtime it is finding a different copy of the file at soversion 6.0.0. I _think_ whatever rpath libchtslib.so was compiled with must not be pointing correctly to the buildtime liblzma.5.dylib.

On 10.11, soversion 6 is the one in /usr/lib and I have soversion 8 through Fink.

**Comment by AndreasHeger** [comment]
Thanks. 

The CI tests work after I have turned of lzma linking by adding ''--disable-lzma" to HTSLIB_CONFIGURE_OPTIONS.

What I am confused by is that the linking works under py3, but fails under py27.

**Comment by nieder** [comment]
Things to check in the py3 builddir and/or install:

1. `otool -L libchtslib.so` to see what version of liblzma.5.dylib it linked against.
2. compare the linker command during the libchtslib.so linking step to see if the py3 build has a different/extra path being added that helps it find the same liblzma.5.dylib at runtime as at buildtime (assuming both build systems have the same 3rd party liblzma.5.dylib (soversion 8) present).

**Comment by jmarshall** [comment]
When compiling Pysam external code, at least on macOS at least some of the time, the compiler does not look in /opt/local/include. Fixed by #829.
